include/demi/types.h

```diff
@@ -148,8 +148,8 @@ extern "C"
         DemiLogLevel_Trace = 5,
     } demi_log_level_t;
 
-    // Logging callback. Arguments are: level, module name, module length, file name, file name length, line number, message, message length, 
-    typedef void (*demi_log_callback_t)(demi_log_level_t, const char*, uint32_t, const char*, uint32_t, uint32_t, const char*, uint32_t);
+    // Logging callback. Arguments are: level, module name, module length, file name, file name length, line number, message, message length,
+    typedef void (*demi_log_callback_t)(demi_log_level_t, const char *, uint32_t, const char *, uint32_t, uint32_t, const char *, uint32_t);
```

Comment:
typedefs for function pointers can have parameter names. That would be much better than using comments to log this info becuase then there is no need to change at 2 places, plus the intent will stand out very clearly at a glance.

Reply:
possibly not for extern typedefs?

Reply:
`extern typedef` is not a standard, so I'm not sure what would happen if we decided to make it `extern` (compiler-dependent). But, I believe it should work regardless, because those parameters wouldn't change the type in anyway and would just server the purpose of crystal clear and maintainable documentation.